log.h

```diff
@@ -20,55 +20,46 @@
 #include <errno.h>
 #include <stdarg.h>
 #include <stdlib.h>
-#include <syslog.h>
 
 #include "macro.h"
 
-#ifdef __cplusplus
-extern "C" {
-#endif
+class Log {
+public:
+    enum class Level {
+        ERROR = 0,
+        WARNING,
+        NOTICE,
+        INFO,
+        DEBUG,
+    };
 
-int log_open(void);
-int log_close(void);
+    static int open();
+    static int close();
 
-int log_get_max_level(void) _pure_;
-void log_set_max_level(int level);
-int log_internal(int level, int error,
-                 const char *file, int line,
-                 const char *format, ...) _printf_format_(5, 6);
+    static Level get_max_level() _pure_ { return _max_level; }
+    static void set_max_level(Level level);
 
-#define log_full_errno(level, error, ...)                               \
-    ({                                                                  \
-         int _level = (level), _e = (error);                            \
-         (log_get_max_level() >= LOG_PRI(_level))                       \
-            ? log_internal(_level, _e, __FILE__, __LINE__, __VA_ARGS__) \
-            : -abs(_e);                                                 \
-        })
+    static void logv(Level level, const char *format, va_list ap);
+    static void log(Level level, const char *format, ...) _printf_format_(2, 3);
```

Comment:
It would be nice, as we are now using a c++ class, to support logging objects, instead of only logging char* strings. I like the approach from std::cout or qLog(), that uses the operator << overriding for logging.
I would like to have something like:

log_debug() << MyObject;

instead of 

log_debug(MyObject.get_string()); //or something similar to that.